src/samVisualize/commands/visualizeSamTemplate/samVisualization.ts

```diff
@@ -0,0 +1,227 @@
+                panel.webview.cspSource
+            )
+        } catch {}
+
+        // To close the visualization if the associated template no longer exists
+        this.disposables.push(
+            vscode.workspace.onDidCloseTextDocument(e => {
+                if (!this.templateDoesExist() && this.webviewPanel) {
+                    panel.dispose()
+                    vscode.window.showErrorMessage('Template associated with visualization closed or renamed')
+                }
+            })
+        )
+
+        const debounceInterval = 1000
+        const debouncedUpdate = _.debounce(this.updateVisualization.bind(this), debounceInterval)
+
+        // Update the visualization on template change
+        this.disposables.push(
+            vscode.workspace.onDidChangeTextDocument(async changedTextDocument => {
+                if (changedTextDocument.document.uri.path === this.textDocumentUri.path) {
+                    debouncedUpdate(changedTextDocument.document.getText())
+                }
+            })
+        )
+
+        const decorationType = vscode.window.createTextEditorDecorationType({
+            backgroundColor: 'rgb(135,206,250,0.3)', // Light sky blue
+        })
+
+        // Handle click events from the webview
+        this.disposables.push(
+            panel.webview.onDidReceiveMessage(message => {
+                // Reveal the template to the left of the webview, unless the webview is in the first column.
+                // In this case, reveal to the right. This way, the template will also reveal beside the webview.
+                const columnToShow =
+                    this.webviewPanel!.viewColumn === vscode.ViewColumn.One
+                        ? vscode.ViewColumn.Two
+                        : this.webviewPanel!.viewColumn!.valueOf() - 1
+
+                vscode.window
+                    .showTextDocument(this.textDocument, { preserveFocus: true, viewColumn: columnToShow })
+                    .then(activeEditor => {
+                        //resource name for clicked resource stored in message.data
+                        const pos = this.resourceLineMap![message.data]
+                        if (pos) {
+                            const decorations: vscode.DecorationOptions[] = []
+                            const startPosition = activeEditor.document.positionAt(pos.start).with({ character: 0 })
+                            const endPosition = activeEditor.document.positionAt(pos.end)
+                            const range = new vscode.Range(startPosition, endPosition)
+                            activeEditor.revealRange(range, vscode.TextEditorRevealType.InCenter)
+                            decorations.push({ range })
+                            activeEditor.setDecorations(decorationType, decorations)
+                        }
+                    })
+            })
+        )
+
+        panel.onDidDispose(() => {
+            this.webviewPanel = undefined
```

Comment:
There's a lot of `!` assertions when dealing with `this.webviewPanel` (and not a lot of try/catches). Is it worth considering making this not settable to `undefined`?

Reply:
Yes, I'll use a boolean to track if it's been disposed or not instead of whether or not the `webviewPanel` is defined